ratis-server/src/main/java/org/apache/ratis/server/impl/RaftServerImpl.java

```diff
@@ -1805,17 +1805,17 @@ TransactionContext getTransactionContext(LogEntryProto entry, Boolean createNew)
   CompletableFuture<Message> applyLogToStateMachine(ReferenceCountedObject<LogEntryProto> nextRef)
       throws RaftLogIOException {
     LogEntryProto next = nextRef.get();
-    if (!next.hasStateMachineLogEntry()) {
-      stateMachine.event().notifyTermIndexUpdated(next.getTerm(), next.getIndex());
-    }
-
-    if (next.hasConfigurationEntry()) {
+    switch (next.getLogEntryBodyCase()) {
+    case CONFIGURATIONENTRY:
       // the reply should have already been set. only need to record
       // the new conf in the metadata file and notify the StateMachine.
       state.writeRaftConfiguration(next);
-      stateMachine.event().notifyConfigurationChanged(next.getTerm(), next.getIndex(), next.getConfigurationEntry());
+      stateMachine.event().notifyConfigurationChanged(next.getTerm(), next.getIndex(),
+          next.getConfigurationEntry());
       role.getLeaderState().ifPresent(leader -> leader.checkReady(next));
-    } else if (next.hasStateMachineLogEntry()) {
+      stateMachine.event().notifyTermIndexUpdated(next.getTerm(), next.getIndex());
```

Comment:
Let move it down to right before `return null`.  Then, we don't have to repeat it for each case.

Reply:
@szetszwo Thank you for the review.

I realized that I repeat it because I missed the point that `StateMachineLogEntry` will return inside the case. I changed this part also for a better readability.